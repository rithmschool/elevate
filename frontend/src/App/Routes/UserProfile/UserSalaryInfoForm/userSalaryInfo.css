.rounded {
  margin: 10% auto;
  background-color: #F4F6F8;
}