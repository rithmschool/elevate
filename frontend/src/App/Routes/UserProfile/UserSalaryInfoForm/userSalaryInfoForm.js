import React, { useState } from "react";
import { Button, Form, InputGroup } from "react-bootstrap";
import "../FormStyles.css";
import "./userSalaryInfoForm.scss";

/** Update user salary */
function UserSalaryInfoForm(props) {
  const [salary, setSalary] = useState(props.salary);
  const [bonus, setBonus] = useState(props.bonus);
  const [equity, setEquity] = useState(props.equity);
  const [isEdit, setIsEdit] = useState(false);

  return (
    <div className="user-salary-form container border shadow rounded mb-5">
      <Form onSubmit={handleSubmit} className="mt-5">
        <div className="form-styles_flex-space-between">
          <h3>Salary info</h3>

          {!isEdit && (
            <i
              data-testid="turnEditOn"
              className="m-3 fas fa-edit fa-1x"
              onClick={() => setIsEdit(true)}
            ></i>
          )}
        </div>

        <Form.Group>
          <Form.Label>Salary</Form.Label>
          <InputGroup>
            <InputGroup.Prepend>
              <InputGroup.Text>$</InputGroup.Text>
            </InputGroup.Prepend>
            <Form.Control
              data-testid="salary"
              onChange={e => setSalary(e.target.value)}
              id="EditUser-salary"
              name="salary"
              type="number"
              step="5000"
              disabled={!isEdit}
              value={salary}
            />
          </InputGroup>
        </Form.Group>

        <Form.Group>
          <Form.Label>Equity</Form.Label>

          <Form.Control
            data-testid="equity"
            onChange={e => setEquity(e.target.value)}
            id="EditUser-equity"
            name="equity"
            type="number"
            step="0.001"
            disabled={!isEdit}
            value={equity}
          />
        </Form.Group>

        <Form.Group>
          <Form.Label>Bonus</Form.Label>
          <InputGroup>
            <InputGroup.Prepend>
              <InputGroup.Text>$</InputGroup.Text>
            </InputGroup.Prepend>
            <Form.Control
              data-testid="bonus"
              onChange={e => setBonus(e.target.value)}
              id="EditUser-bonus"
              name="bonus"
              type="number"
              step="500"
              disabled={!isEdit}
              value={bonus}
            />
          </InputGroup>
        </Form.Group>

        <div className="row justify-content-center">
          {isEdit && (
            <div>
              <Button
                id="userSalaryBtn"
                className="login-submit mr-3 ml-3"
                type="submit"
              >
                Submit
              </Button>

              <h6
                data-testid="cancel"
                className="mr-3 ml-3 form-styles_cancel"
                onClick={() => setIsEdit(false)}
              >
                Cancel
              </h6>
            </div>
          )}
        </div>
      </Form>
    </div>
  );

  function handleSubmit(evt) {
    evt.preventDefault();

    const updatedSalary = {
      user_id: props.user_id,
      salary,
      bonus,
      equity
    };

    props.handleSalaryUpdate(updatedSalary);
    setIsEdit(false);
  }
}

export default UserSalaryInfoForm;
