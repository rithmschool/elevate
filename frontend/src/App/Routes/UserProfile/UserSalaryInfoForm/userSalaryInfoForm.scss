.user-salary-form {
  .rounded {
    margin: 10% auto;
    background-color: #f4f6f8;
  }
}
