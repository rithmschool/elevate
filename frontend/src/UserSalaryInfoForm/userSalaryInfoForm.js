import React from "react";
import { Button, Form } from "react-bootstrap";
import "../FormStyles.css";

/** Update user salary */
class UserSalaryInfoForm extends React.Component {
  
  constructor(props){
    super(props);

    this.state = {
      user_id: this.props.user_id,
      salary: this.props.salary, 
      bonus: this.props.bonus,
      equity: this.props.equity,
      isEdit: false,
    }

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.toggleEditForm = this.toggleEditForm.bind(this)
  }

  toggleEditForm() {
    this.setState(state => ({isEdit: !state.isEdit }));
  }

  handleChange(evt){
		this.setState({ [evt.target.name]: evt.target.value });
  }
  
  handleSubmit(evt){
    evt.preventDefault()
   
    const salary = {
      user_id: this.state.user_id,
      salary: this.state.salary,
      bonus: this.state.bonus,
      equity: this.state.equity
    };

    this.props.handleSalaryUpdate(salary);
    this.setState({isEdit: false})
  }
  
  render() {
    const isEdit = this.state.isEdit

    return (
      <div 
        className="EditPUserForm container border rounded"
        style={{backgroundColor:'#F4F6F8'}}
      >

        <div className="form-inside-container mt-5">
          <Form onSubmit={ this.handleSubmit }> 
            <div className="form-styles_flex-space-between">
              <h3>Salary info</h3>

              {!isEdit &&
                <i className="m-3 fas fa-edit fa-1x"
                  onClick={this.toggleEditForm}>
                </i>}
            </div>

            <Form.Group>
              <span>First name</span>

              <Form.Control
                onChange={this.handleChange}
                id="EditUser-salary"
                name="salary"
                type="number"
                step="5000"
                disabled={!isEdit}  
                value={this.state.salary}
              />
            </Form.Group>

            <Form.Group>
              <span>Equity</span>

              <Form.Control
                onChange={this.handleChange}
                id="EditUser-equity"
                name="equity"
                type="number"
                step="0.001"
                disabled={!isEdit}  
                value={this.state.equity}
              />
            </Form.Group>

            <Form.Group>
              <span>Bonus</span>

              <Form.Control
                onChange={this.handleChange}
                id="EditUser-bonus"
                name="bonus"
                type="number"
                step="500"
                disabled={!isEdit}  
                value={this.state.bonus}
              />
            </Form.Group>

            <div className="row justify-content-center">
              {isEdit &&
                <div>
                  <Button
                    className="login-submit mr-3 ml-3"
                    type="submit">
                    Submit
                  </Button>

                  <h6 
                    className="mr-3 ml-3 form-styles_cancel"
                    onClick={this.toggleEditForm}>
                    Cancel
                  </h6>
                </div>}
            </div>
          </Form>
        </div>
      </div>
    )
  }
}

export default UserSalaryInfoForm;
