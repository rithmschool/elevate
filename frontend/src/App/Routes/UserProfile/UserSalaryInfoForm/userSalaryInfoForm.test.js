import React from "react";
import Enzyme from "enzyme";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import UserSalaryInfoForm from "./userSalaryInfoForm";
import { Form } from "react-bootstrap";
import { render, fireEvent } from "@testing-library/dom";

describe("UserSalaryInfoForm", function() {
  let wrapper;

  let latestSalary = {
    user_id: 1,
    salary: 50000,
    bonus: 100,
    equity: 0.1
  };

  beforeEach(() => {
    wrapper = mount(<UserSalaryInfoForm {...latestSalary} />);
  });

  it("renders without crashing", function() {
    shallow(<UserSalaryInfoForm {...latestSalary} />);
  });

  it("shows my default text", () => {
    expect(wrapper.find("h3").text()).toEqual("Salary info");
  });

  it("matches snapshot", function() {
    let wrapper = shallow(<UserSalaryInfoForm />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it("renders all inputs", function(){
    expect(wrapper.find(Form.Control)).toHaveLength(3);
  })
});
