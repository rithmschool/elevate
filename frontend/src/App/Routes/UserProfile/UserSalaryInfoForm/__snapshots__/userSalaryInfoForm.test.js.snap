// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSalaryInfoForm matches snapshot 1`] = `
<UserSalaryInfoForm
  bonus={100}
  equity={0.1}
  salary={50000}
  user_id={1}
>
  <div
    className="EditPUserForm container border rounded"
    style={
      Object {
        "backgroundColor": "#F4F6F8",
      }
    }
  >
    <div
      className="form-inside-container mt-5"
    >
      <Form
        inline={false}
        onSubmit={[Function]}
      >
        <form
          className=""
          onSubmit={[Function]}
        >
          <div
            className="form-styles_flex-space-between"
          >
            <h3>
              Salary info
            </h3>
            <i
              className="m-3 fas fa-edit fa-1x"
              data-testid="turnEditOn"
              onClick={[Function]}
            />
          </div>
          <FormGroup>
            <div
              className="form-group"
            >
              <span>
                Salary
              </span>
              <FormControl
                data-testid="salary"
                disabled={true}
                id="EditUser-salary"
                name="salary"
                onChange={[Function]}
                step="5000"
                type="number"
                value={50000}
              >
                <input
                  className="form-control"
                  data-testid="salary"
                  disabled={true}
                  id="EditUser-salary"
                  name="salary"
                  onChange={[Function]}
                  step="5000"
                  type="number"
                  value={50000}
                />
              </FormControl>
            </div>
          </FormGroup>
          <FormGroup>
            <div
              className="form-group"
            >
              <span>
                Equity
              </span>
              <FormControl
                data-testid="equity"
                disabled={true}
                id="EditUser-equity"
                name="equity"
                onChange={[Function]}
                step="0.001"
                type="number"
                value={0.1}
              >
                <input
                  className="form-control"
                  data-testid="equity"
                  disabled={true}
                  id="EditUser-equity"
                  name="equity"
                  onChange={[Function]}
                  step="0.001"
                  type="number"
                  value={0.1}
                />
              </FormControl>
            </div>
          </FormGroup>
          <FormGroup>
            <div
              className="form-group"
            >
              <span>
                Bonus
              </span>
              <FormControl
                data-testid="bonus"
                disabled={true}
                id="EditUser-bonus"
                name="bonus"
                onChange={[Function]}
                step="500"
                type="number"
                value={100}
              >
                <input
                  className="form-control"
                  data-testid="bonus"
                  disabled={true}
                  id="EditUser-bonus"
                  name="bonus"
                  onChange={[Function]}
                  step="500"
                  type="number"
                  value={100}
                />
              </FormControl>
            </div>
          </FormGroup>
          <div
            className="row justify-content-center"
          />
        </form>
      </Form>
    </div>
  </div>
</UserSalaryInfoForm>
`;
